### ISSUE TITLE
Method reference type inference warning given despite lack of inference

### ISSUE THREAD

**Issue by astiob** [typetools/checker-framework#1319]
This code produces a `methodref.inference.unimplemented` warning (as in #802):

```java
java.util.stream.Stream<String> stream = java.util.stream.Stream.of("foo");
Iterable<String> iterable = stream::iterator;
```

```
warning: [methodref.inference.unimplemented] This version of the Checker Framework does not type-check method references with implicit type arguments.
Iterable<String> iterable = stream::iterator;
                            ^
```

(It does not matter what the stream’s actual element type is or how the stream is obtained.)

I don’t understand how this warning is relevant to this line or why a warning is given at all.

I’ve tried Checker Framework 2.1.12 and 2.1.11, both with Oracle’s javac 1.8.0_40.
